z3/src/solver.rs

```diff
@@ -213,10 +222,17 @@ impl<'ctx> Solver<'ctx> {
         })
     }
 
-    /// Set the current solver using the given parameters.
-    pub fn set_params(&self, params: &Params<'ctx>) {
-        let guard = Z3_MUTEX.lock().unwrap();
-        unsafe { Z3_solver_set_params(self.ctx.z3_ctx, self.z3_slv, params.z3_params) };
```

Comment:
I don't think we should remove `set_params` here.

Reply:
Oh, I messed up with the rebase. I can fix it tomorrow.